### PR TITLE
simplify path handling in `jj sparse set --edit`

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/commands/sparse.rs [hunk 1]
```diff
@@ -111,14 +110,7 @@ fn cmd_sparse_set(
...
         .try_collect()?;
-    // Determine inputs of `edit` operation now, since `workspace_command` is
-    // inaccessible while the working copy is locked.
-    let edit_inputs = args.edit.then(|| {
-        (
-            workspace_command.repo().clone(),
-            workspace_command.workspace_root().clone(),
-        )
-    });
+    let repo_path = workspace_command.repo().repo_path().to_owned();
```
Comment (martinvonz) [inline cli/src/commands/sparse.rs:115]: I think we should also change line 90 to print workspace-relative paths

**Review by martinvonz — COMMENTED** [review]
Thanks. I think I've heard from several internal users that they found the cwd-relative paths confusing.

File: cli/src/commands/sparse.rs
[the remark this answers is not in the document]
Comment (yuja) [reply cli/src/commands/sparse.rs:1]: Do you mean we should also change `--add` (and `list`) to use workspace-relative paths? I won't disagree about that, but I would also want to extract `sparse edit` subcommand if we introduce major breaking change there.

This PR is minor cleanup that drops support of `foo/../bar` or `/path/to/ws/foo` from `sparse set --edit`. "edit" was workspace-relative from the start.

[the remark this answers is not in the document]
Comment (martinvonz) [reply cli/src/commands/sparse.rs:1]: Yes, that's what I meant, but I didn't realize that `--edit` was always workspace-relative. That's clearly out of scope for this PR then :)

**Review by martinvonz — APPROVED** [review]

File: cli/src/commands/sparse.rs
[the remark this answers is not in the document]
Comment (yuja) [reply cli/src/commands/sparse.rs:1]: Ok, I'll send a follow up.

**Pull request merged**